offer-manager-domain/src/main/groovy/life/qbic/business/offers/Offer.groovy

```diff
@@ -484,9 +486,14 @@ class Offer {
 
     private BigDecimal discountAmountForProductItem(ProductItem productItem) {
         BigDecimal discount = 0
+        MathContext rounding = new MathContext(2, RoundingMode.CEILING)
         if (productItem.product instanceof PrimaryAnalysis
                 || productItem.product instanceof SecondaryAnalysis) {
-            discount = quantityDiscount.apply(productItem.quantity as Integer, calculateItemNet(productItem))
+            BigDecimal unitPrice = selectedCustomerAffiliation.category == AffiliationCategory.INTERNAL ? productItem.product.internalUnitPrice : productItem.product.externalUnitPrice
```

Comment:
```suggestion
            BigDecimal unitPrice = selectedCustomerAffiliation.category == AffiliationCategory.INTERNAL ? productItem.product.internalUnitPrice : productItem.product.externalUnitPrice
```
you could extract a method here since the same logic is used in other places here

Reply:
Done:)